### PR TITLE
Switch to Agency

### PR DESCRIPTION
**Pull request by lchanmann**
~Stub thread pool would not execute context agent upon submission. To actually run the submitted agents, the client needs to call `invokeAll()` manually.~

## Changes

- Replace ThreadPool with Agency
- Rename FixedThreadPool to ThreadPoolAgency
- Add AgentAgency (it was StubThreadPool before)
- Update "threadPool" service setting

### CODE REVIEW

**Comment by jlhughes** [comment]
Who (or will) calls invokeAll?

**Comment by lchanmann** [comment]
> Who (or will) calls invokeAll?

I plan to use it for IdentityMind integration test. It allows us to invoke the real (async) actions before performing the assertions. For example:

```java
((IdentityMindService) getIdentityMindService()).setDefaultProfile("ACCEPT");
Transaction tx = newTransaction(1000);
tx = (Transaction) ((DAO) getTransactionDAO()).put(tx);

test(findTxApprovalRequest(tx) == null,
  "No approval request is created yet before the IdentityMind transaction rule is run");

// Invoke IdentityMind validator
((StubThreadPool) getThreadPool()).invokeAll();

// Assertions
ApprovalRequest found = findTxApprovalRequest(tx);
test(found != null && found.getApprover() == getIdentityMindUserId() && found.getStatus() == ApprovalStatus.APPROVED,
  "An approval request with status=APPROVED is created when the IdentityMind ACCEPT-ed the transaction");

tx = (Transaction) ((DAO) getTransactionDAO()).find(tx);
test(tx.getStatus() == TransactionStatus.COMPLETED, "Compliance transaction should be COMPLETED");
```

File: src/foam/nanos/pool/ThreadPool.java [hunk 1]
```diff
@@ -10,5 +10,5 @@
 import foam.core.X;
 
 public interface ThreadPool {
```
Comment (kgrgreer) [inline src/foam/nanos/pool/ThreadPool.java]: Isn't this just the same as Agency?

**Comment by kgrgreer** [comment]
Isn't threadpool the same interface as Agency and FixedThreadPool should just be an Agency implementation?

**Comment by lchanmann** [comment]
Added your suggestions. @kgrgreer

File: src/foam/nanos/pool/AgentAgency.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  Queue<Runnable> agents = new LinkedList<>();
+
+  @Override
+  public void submit(X x, ContextAgent agent, String description) {
+    agents.add(new ContextAgentRunnable(x, agent, description));
+  }
+
+  public void execute() {
+    while ( ! agents.isEmpty() ) {
+      agents.poll().run();
```
Comment (kgrgreer) [inline src/foam/nanos/pool/AgentAgency.java]: Why poll?

Comment (lchanmann) [reply src/foam/nanos/pool/AgentAgency.java]: Fixed.

**Pull request merged**